src/commands/miscellaneous/ping.ts

```diff
@@ -3,45 +3,44 @@ import { Message } from 'discord.js';
 import {
   CodeyCommand,
   CodeyCommandDetails,
-  CodeyCommandResponseType,
   SapphireMessageExecuteType,
-  SapphireMessageRequest,
   SapphireMessageResponse,
 } from '../../codeyCommand';
 
-const initialPingContent = 'Ping?';
-
-const getApiLatency = (
-  initialPing: SapphireMessageRequest,
-  messageFromUserAsMessage: Message<boolean>,
-) => {
-  const initialPingAsMessage = <Message<boolean>>initialPing;
-  return initialPingAsMessage.createdTimestamp - messageFromUserAsMessage.createdTimestamp;
-};
-
-const content = (botLatency: number, apiLatency: number) =>
-  `Pong from JavaScript! Bot Latency ${botLatency}ms. API Latency ${apiLatency}ms.`;
-
-const executeCommand: SapphireMessageExecuteType = async (
+const executePingCommand: SapphireMessageExecuteType = async (
   client,
   messageFromUser,
   _args,
-  initialMessageFromBot,
 ): Promise<SapphireMessageResponse> => {
-  // Assert message types are Message<boolean>
-  // We have to do this because APIMessage does not have "createdTimestamp" property
-  const messageFromUserAsMessage = <Message<boolean>>messageFromUser;
   const botLatency = client.ws.ping;
-  if (initialMessageFromBot) {
-    const apiLatency = getApiLatency(initialMessageFromBot, messageFromUserAsMessage);
-    return new Promise((resolve, _reject) => resolve(content(botLatency, apiLatency)));
+
+  // we can freely coerce to a Message because the alternative type: APIMessage
+  // is only received:
+  // "when you receive an interaction from a guild that the client is not in."
+  // src: https://github.com/discordjs/discord.js/issues/7001#issuecomment-972422214
+  const initialResponse = <Message<boolean>>await messageFromUser.reply({
+    content: 'Pong?',
+    ephemeral: pingCommandDetails.isCommandResponseEphemeral,
+    fetchReply: true,
+  });
+
+  const apiLatency = initialResponse.createdTimestamp - messageFromUser.createdTimestamp;
+  const stringReply = content(botLatency, apiLatency);
+
+  // have to modify the original interaction when using slash commands
+  // because you cannot edit ephemeral responses.
```

Comment:
What does this comment mean exactly? Aren't you currently editing the ephemeral response?

Reply:
I am editing it, but it's happening from the interaction itself, not from the response

Reply:
I'm confused what you mean by this - what do you mean that the editing is coming from the interaction and not from the response?

Reply:
instead of editing the reply that we're sending (`initialRespone`) directly via `initialResponse.edit`, we instead have to use the `messageFromUser` (a slash command `ChatInputInteraction`) and call the`editReply` method on it. We know `messageFromUser` must be a slash command in the second case because only slash commands can proc ephemeral messages
```js
     await messageFromUser.editReply(stringReply);
```

Reply:
I see. Perhaps do you mind editing the comment to "have to use the original interaction to edit the reply when using slash commands because an ephemeral reply cannot be directly edited" to make it more clear?

Reply:
Done in [037362c](https://github.com/uwcsc/codeybot/pull/329/commits/037362c6b847fcb0f9412c923200d7046956bd75)